### PR TITLE
Third integration of type definitions for the decomposition tool

### PR DESCRIPTION
**Pull request by zhululai**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files

---

## Short Description

This pull request includes the type definitions needed to create the toy examples of monolithic and microservice applications that can be decomposed by the decomposition tool.

## Resolving Issue / Feature

In particular, the following new features have been enabled by the pull request:
1. Specification of web and container applications
2. Support of direct invocations to serverless functions

A few errors were also fixed when the upstream was merged into the fork.

### CODE REVIEW

File: nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@
           --function-name {{ function_name }}
           --reserved-concurrent-executions {{ lambda_concurrency }}
           --region {{ aws_region }}
-      when: (lambda_concurrency is defined) and (lambda_concurrency != 0)
```
Comment (miwurster) [inline nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml]: Have you tested the deployment after this change? ... also when concurrency is disabled? I remember that we had some issues with xOpera whiteout this addition. @v-yussupov what do you think?

File: nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/files/create/create.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@
           --function-name {{ function_name }}
           --reserved-concurrent-executions {{ lambda_concurrency }}
           --region {{ aws_region }}
-      when: (lambda_concurrency is defined) and (lambda_concurrency != 0)
```
Comment (miwurster) [inline nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/files/create/create.yml]: Same as above

File: nodetypes/tosca.nodes/WebApplication/NodeType.tosca [hunk 3]
```diff
@@ -10,6 +10,7 @@ node_types:
     properties:
       context_root:
         type: string
```
Comment (miwurster) [inline nodetypes/tosca.nodes/WebApplication/NodeType.tosca:12]: We should avoid changing the normative types. We create them according to the specification in the standard.

**Review by miwurster — CHANGES_REQUESTED** [review]
Very nice contribution. However I have some doubts regarding the changes in the `create.yml` files below.

File: nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml (hunk 1 above)
Comment (miwurster) [reply nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml]: ping @v-yussupov

Comment (v-yussupov) [reply nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml]: yes, the ansible task fails if `lambda_concurrency` is absent, which is the case when modelers do not define this optional property in templates.

Comment (zhululai) [reply nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml]: No, I think this looks like a bug of xOpera, which should be able to deal with unspecified optional properties. I will revert this change in the pull request at present, but we need to discuss this issue with XLAB perhaps in the next WP4 call. The decomposition tool doesn't recognize a concurrency of zero, which in my opinion is a bit confusing to the users.

File: nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/files/create/create.yml (hunk 2 above)
Comment (zhululai) [reply nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/files/create/create.yml]: Same as above

File: nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml (hunk 1 above)
Comment (miwurster) [reply nodetypes/radon.nodes.aws/AwsLambdaFunction/files/create/create.yml]: Once this is reverted we can merge it. Thanks.

File: nodetypes/tosca.nodes/WebApplication/NodeType.tosca (hunk 3 above)
Comment (zhululai) [reply nodetypes/tosca.nodes/WebApplication/NodeType.tosca:12]: I think this is a mistake in the TOSCA specification. The properties table is not consistent to the node type definition in [[1]](https://docs.oasis-open.org/tosca/TOSCA-Simple-Profile-YAML/v1.3/os/TOSCA-Simple-Profile-YAML-v1.3-os.html#DEFN_TYPE_NODES_WEBAPPLICATION), so I refer to the intepretation by the TOSCA parser of OpenStack in [[2]](https://github.com/openstack/tosca-parser/blob/master/toscaparser/elements/TOSCA_definition_1_0.yaml).

Comment (miwurster) [reply nodetypes/tosca.nodes/WebApplication/NodeType.tosca:12]: Well, considering tool-specific interpretation is fine by me. But then I would encourage you to consider the interpretation by xOpera. In the end, they need to support the models at runtime and, furthermore, are compliant with TOSCA 1.3 (the OpenStack parser only 1.0 AFAIK).

Comment (zhululai) [reply nodetypes/tosca.nodes/WebApplication/NodeType.tosca:12]: xOpera merely interpretates the node type definition, so xOpera if the property is required or not. What matters is the underlying Ansible playbooks, which is not currently not available. I checked the history of the TOSCA specification. This part doesn't change from TOSCA 1.0 to 1.3. The reason why I refer to the OpenStack parser is that it is widely used for a long time.

Comment (zhululai) [reply nodetypes/tosca.nodes/WebApplication/NodeType.tosca:12]: xOpera merely interpretates the node type definition, so it doesn't care if the property is required or not. What matters is the underlying Ansible playbooks, which are currently not available. I checked the history of the TOSCA specification. This part doesn't change from TOSCA 1.0 to 1.3. The reason why I refer to the OpenStack parser is that it is widely used for a long time.

**Review by miwurster — APPROVED** [review]

**Pull request merged**